docs/architecture/adr-017-single-binary-upgrade.md

```diff
@@ -0,0 +1,141 @@
+# ADR 017: Single binary upgrade
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/22: initial draft
+
+## Context
+
+It important to sync from genesis with one binary.
+It is needed for wallets, block explorers, indexers, analytics (e.g. Nansen).
+It makes it possible to sync headers from chain and download blobs from third party service.
+
+## Alternative Approaches
+
+Instead of having a single binary, we can have multiple binaries that are managed by another supervisory binary e.g. [Cosmovisor](https://docs.desmos.network/fullnode/cosmovisor/).
+Osmosis does it through having a significant number of binaries gathered in one docker container and switch them in particular heights.
+This approach has a lot of maintenance burden and is fragile to sync from genesis.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+- Cosmos blocks contain a version.app (version of state machine) that can be used to indicate which version of `celestia-app` should be used to process a block.
+  - Example: Osmosis is currently on [version.app: 15](https://www.mintscan.io/osmosis/blocks/8804111) and release [v15.0.0](https://github.com/osmosis-labs/osmosis/releases/tag/v15.0.0).
+
+  - Have to maintain old versions of binaries. Need to update dependencies for security issues.
+
+- Genesis has an original version.app. Does celestia-app need to explicitly define this?
+
+- Bitcoin / Ethereum leverage if statement upgrade logic. Downside to if statement logic in Cosmos ecosystem is if statements may not be relevant for new chains
+
+- UX issue to improve experience for node operators to upgrade
+
+- Maintenance issue for maintaining historical releases of binaries
```

Comment:
Maybe we can articulate these as the two compelling points driving this ADR. So proposal to life them to the top as part of a problem statement section